src/Service/GatewayResourceService.php

```diff
@@ -142,33 +142,48 @@ public function getSource(string $reference, string $pluginName): ?Source
     }//end getSource()
 
     /**
-     * Find all sources that have a location that match the specified url.
+     * Find a source that has a location that match the specified url.
      * Todo: we should use a mongoDB filter instead of this, sources should exist in MongoDB.
-     * Todo: for future reference, there is a function with very similar BL in the CustomerInteractionBundle->CustomerInteractionService->getSource(), we could/should merge that code with this function.
+     * This function gets used in the CustomerInteractionBundle, Gateway->synchronizationService.
      *
-     * @param string $url        The url we are trying to find a matching source for.
-     * @param string $pluginName The name of the plugin that requests these resources.
+     * @param string      $url        The url we are trying to find a matching source for.
+     * @param string      $pluginName The name of the plugin that requests these resources.
+     * @param string|null $endpoint   The resulting endpoint (the remainder of the path).
      *
-     * @return array|null
+     * @return Source|null The source found or null.
      */
-    public function findSourcesForUrl(string $url, string $pluginName): ?array
+    public function findSourceForUrl(string $url, string $pluginName, ?string &$endpoint = null): ?Source
```

Comment:
Have you checked where this function is used before renaming it?

Reply:
Yess, so far i know/found only in CustomerInteractionBundle and Gateway->synchronizationService.